ddl/table.go

```diff
@@ -63,13 +70,31 @@ func onCreateTable(d *ddlCtx, t *meta.Meta, job *model.Job) (ver int64, _ error)
 			// TODO: Add restrictions to this operation.
 			go splitTableRegion(d.store, tbInfo.ID)
 		}
-		// Finish this job.
-		job.FinishTableJob(model.JobStateDone, model.StatePublic, ver, tbInfo)
-		asyncNotifyEvent(d, &util.Event{Tp: model.ActionCreateTable, TableInfo: tbInfo})
-		return ver, nil
+	case model.StateWriteReorganization:
+		// reorganization -> public (insert data before we make the table public)
+		err = doCreateTableInsert(d, t, job, tbInfo, snapshotTS)
+		if err != nil {
+			job.State = model.JobStateCancelled
+			return ver, errors.Trace(err)
+		}
+
+		tbInfo.State = model.StatePublic
+		job.SchemaState = model.StatePublic
 	default:
 		return ver, ErrInvalidTableState.GenWithStack("invalid table state %v", tbInfo.State)
 	}
+	tbInfo.UpdateTS = t.StartTS
+	ver, err = updateVersionAndTableInfo(t, job, tbInfo, originalState != tbInfo.State)
+	if err != nil {
+		job.State = model.JobStateCancelled
```

Comment:
No needs to set job.State to model.JobStateCancelled, you can just return the error, then ddl will retry this job.

Reply:
Got it